compiler/erg_common/style.rs

```diff
@@ -0,0 +1,512 @@
+        (self.colors.warning, self.characters.line)
+    }
+
+    pub const fn exception(&self) -> (Color, char) {
+        (self.colors.exception, self.characters.wave)
+    }
+
+    pub const fn hint(&self) -> (Color, char) {
+        (self.colors.hint, self.characters.wave)
+    }
+
+    pub fn error_kind_format(&self, kind: &str, err_num: usize) -> String {
+        self.characters.error_kind_format(kind, err_num)
+    }
+}
+
+pub const THEME: Theme = Theme {
+    colors: COLORS,
+    characters: CHARS,
+};
+
+#[cfg(not(feature = "unicode"))]
+pub const CHARS: Characters = Characters {
+    hat: '-',
+    line: '-',
+    vbar: '|',
+    wave: '~',
+    lbot: '`',
+    vbreak: ':',
+    lbrac: '[',
+    rbrac: ']',
+};
+
+#[cfg(feature = "unicode")]
+pub const CHARS: Characters = Characters {
+    hat: '-',
+    line: '─',
+    vbar: '│',
+    wave: '~',
+    lbot: '╰',
+    vbreak: '·',
+    lbrac: '[',
+    rbrac: ']',
+};
+
+#[cfg(not(feature = "pretty"))]
+pub const COLORS: ThemeColors = ThemeColors {
+    error: Color::Red,
+    warning: Color::Yellow,
+    exception: Color::Magenta,
+    gutter: Color::Cyan,
+    hint: Color::Green,
+};
+
+#[cfg(feature = "pretty")]
+pub const COLORS: ThemeColors = ThemeColors {
+    error: Color::CustomRed,
+    warning: Color::CustomYellow,
+    exception: Color::CustomMagenta,
+    gutter: Color::CustomCyan,
+    hint: Color::CustomGreen,
+};
+
+///
+/// `StrSpan` is for const color and attribute &str.
+/// It is an immutable string.
+/// # Example
+/// ```
+/// const URL: StrSpan = StrSpan::new(
+///    "https://github.com/erg-lang/erg",
+///    Some(Color::White),
+///    Some(Attribute::Underline),
+/// );
+/// ```
+#[derive(Debug)]
+pub struct StrSpan<'a> {
+    span: &'a str,
+    color: Option<Color>,
+    attribute: Option<Attribute>,
+}
+
+impl<'a> StrSpan<'a> {
+    pub const fn new<'b: 'a>(
+        span: &'b str,
+        color: Option<Color>,
+        attribute: Option<Attribute>,
+    ) -> Self {
+        Self {
+            span,
+            color,
+            attribute,
+        }
+    }
+}
+
+impl std::fmt::Display for StrSpan<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match (self.color, self.attribute) {
+            (None, None) => todo!(),
+            (None, Some(attr)) => write!(f, "{}{}{}", attr.as_str(), self.span, ATTR_RESET),
+            (Some(color), None) => write!(f, "{}{}{}", color.as_str(), self.span, RESET),
+            (Some(color), Some(attr)) => {
+                write!(
+                    f,
+                    "{}{}{}{}{}",
+                    color.as_str(),
+                    attr.as_str(),
+                    self.span,
+                    RESET,
+                    ATTR_RESET
+                )
+            }
+        }
+    }
+}
+
+///
+/// `StringSpan` is for coloring and attribute text.
+/// String, Color(&str) and Attribute(&str)
+///
+#[derive(Debug)]
+pub struct StringSpan {
```

Comment:
I think the mean of `Dpan` is unclear.
Wouldn't `DecoratedString` be better?

Reply:
Or `StyledString`.

Reply:
This `Span` comes from [html's span](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span)
The word Span has no meaning and the word `Span` itself means modified text

Reply:
I think `StringSpan` can be misinterpreted as a string having information about its position and span in the code.

Reply:
I certainly didn't take that into account.
I'll change `StringSpan' to `StyledString`